pubsub/google/cloud/pubsub_v1/subscriber/policy/thread.py

```diff
@@ -330,10 +335,47 @@ def on_response(self, response):
 
         For each message, schedule a callback with the executor.
         """
-        for msg in response.received_messages:
+        if self._ack_on_receipt:
```

Comment:
This two separate functions thing is really confusing. The names mean basically nothing, and the bulk of the body is copied and pasted. Even if we had the knob, why is it not just:

```
if knob:
    knob_only_thing
other_things
```